pennylane/transforms/qcut.py

```diff
@@ -843,3 +846,331 @@ def qcut_processing_fn(
+            Optional, defaults to ``max_free_gates``.
+        imbalance_tolerance (float): The global maximum allowed imbalance for all partition trials.
+            Optional, defaults to unlimited imbalance. Used only if there's a known hard balacing
+            constraint on the partitioning problem.
+
+    **Example**
+
+    .. code-block:: python
+
+        import pennylane as qml
+
+        dev_a = qml.device('default.qubit', wires=4)
+        dev_b = qml.device('default.qubit', wires=6)
+
+        with qml.tape.QuantumTape() as tape:
+            qml.RX(0.4, wires=0)
+            qml.RY(0.9, wires=0)
+            qml.CNOT(wires=[0, 1])
+            qml.expval(qml.PauliZ(1))
+
+        tape_dag = qml.transforms.tape_to_graph(tape)
+
+    The following cut strategy specifies that a circuit should be cut into between
+    ``2`` to ``5`` fragments, with each fragment having at most ``6`` wires and
+    at least ``4`` wires:
+
+    >>> cut_strategy = qml.transforms.CutStrategy(
+    ...     max_free_wires=6,
+    ...     min_free_wires=4,
+    ...     num_fragments_probed=(2, 5),
+    ... )
+
+    """
+
+    # pylint: disable=too-many-arguments, too-many-instance-attributes
+
+    devices: InitVar[Sequence[qml.Device]] = None
+    max_free_wires: int = None
+    min_free_wires: int = None
+    num_fragments_probed: Union[int, Sequence[int]] = None
+    max_free_gates: int = None
+    min_free_gates: int = None
+    imbalance_tolerance: float = None
+
+    HIGH_NUM_FRAGMENTS: ClassVar[int] = 20
+    HIGH_PARTITION_ATTEMPS: ClassVar[int] = 20
+
+    def __post_init__(
+        self,
+        devices,
+    ):
+        """Deriving cutting constraints from given devices and parameters."""
+
+        self.max_free_wires = self.max_free_wires or self.min_free_wires
+        if isinstance(self.num_fragments_probed, int):
+            self.num_fragments_probed = [self.num_fragments_probed]
+        if isinstance(self.num_fragments_probed, (list, tuple)):
+            self.num_fragments_probed = sorted(self.num_fragments_probed)
+            self.k_lower = self.num_fragments_probed[0]
+            self.k_upper = self.num_fragments_probed[-1]
+            if self.k_lower <= 0:
+                raise ValueError("`num_fragments_probed` must be positive int(s)")
+        else:
+            self.k_lower, self.k_upper = None, None
+
+        if devices is None and self.max_free_wires is None:
+            raise ValueError("One of arguments `devices` and max_free_wires` must be provided.")
+
+        if isinstance(devices, qml.Device):
+            devices = (devices,)
+
+        if devices is not None:
+            if not isinstance(devices, (list, tuple)) or any(
+                (not isinstance(d, qml.Device) for d in devices)
+            ):
+                raise ValueError(
+                    f"Argument `devices` contain at least one `Device` instance, got {type(devices)}."
```

Comment:
Is this the right error message? We raise an error if:
- Devices is not a list of tuple, or
- Any element in device is not a `qml.Device`

Maybe the error message should just read:
```
Argument `devices` must be a list or tuple containing elements of type qml.Device
```
(Also, should we just check for `Sequence` rather than list/tuple?)

Reply:
Ya definitely screwed up that sentence, switched to your suggestion.